[Den] css 변수 구현 / DOM API 학습 및 활용 예정

sjuhan123 opened the pull request:
## 학습 계획

### 피드백 반영

- [x] : css 변수 적용 예정  
  - [x] : root 클래스에 font style 변수화
  - [x] : css 하드코딩된 속성들 변수로 지정
- [ ] : 클래스명 전체적으로 점검
  - [x] : header 영역 클래스명 점검
  - [x] : sidebar 영역 클래스명 점검
  - [ ] : modal 오브제들 클래스명 점검
- [x] : 전역변수를 지역변수로 만들어보기
  - [x]  : 기능별로? 혹은 역할별로 함수 혹은 클래스로 옮기기

sjuhan123 committed:
feat. 사이드바 레이아웃 및 애니메이션 구현

extra menu 레이아웃 및 애니메이션 구현
right menu 레이아웃 및 애니메이션 구현

sjuhan123 committed:
feat. sidebar 애니매이션 구현 완료

모두보기 버튼 클릭시 extra 메뉴 오픈 애니매이션 구현 완료

sjuhan123 committed:
docs. Week1 day4 학습계획 이행완료 목록 체크

sjuhan123 committed:
docs. Wee1 day5 학습계획 이행목록 체크

sjuhan123 committed:
docs. Week2 day1 학습계획 작성

sjuhan123 committed:
docs. Week2 학습 계획 작성 및 리드미 파일 전반적인 내용 수정

sjuhan123 committed:
refactor. css font style css variuables를 이용해서 리팩토링 진행

sjuhan123 committed:
refactor. html, css 리팩토링 진행

html 필요없는 container 역할 태그 제거
html 클래스명 변경
css 스타일 border, margin 등 전체적으로 리팩토링 진행

crongro commented:
> 피드백 반영
> * css 변수 적용 예정
> * 클래스명 전체적으로 점검
> * js 코드 리팩토링 예정 : DOM API 활용 sidebar 구현

네 피드백과 이번주 미션위주로 계획을 세워서 해보세요.

sjuhan123 committed:
refactor. header 영역 레이아웃 변경, js 코드 리팩토링 진행

header 영역 불필요한 태그 삭제
header 영역 불필요한 속성 삭제
header 영역 요소 클래스명 통일화, css 통일화
header 영역 버튼 요소에 hover 적용 - border 라인
header, sidebar 영역 관련 js 코드 전체적으로 리팩토링 진행

sjuhan123 committed:
feat. sidebar 서브 메뉴 구현

sjuhan123 committed:
docs. 수요일 학습계획 작성

crongro commented on index.js:
```diff
@@ -1,57 +1,142 @@
-
-modalLayouts.forEach((node) => {
-  node.addEventListener("mouseover", () => {
+function sideBarHandler(findNodeModule, dimLayer){
```
이 정도 함수는 비교적 너무 크긴해요. 
더 작은 부분으로 분리해보면 좋겠네요. 

어떤 부분을 분리할지 생각해보세요.

crongro commented on index.js:
```diff
@@ -1,57 +1,142 @@
     dimLayer.style.display = 'block';
+  });
+  sideBarLayer.addEventListener('click', (evt) => {
+    const ontarget = evt.target;
```
해체할당 문법을 사용해보세요.

crongro commented on index.js:
```diff
@@ -1,57 +1,142 @@
+      dimLayer.style.display = 'none';
+    }
+  
+    if(name === 'showall') {
```
메시지 문자열이라면 SHOW_ALL 이렇게 대문자 표현방법도 자주 사용합니다.

crongro commented on index.js:
```diff
@@ -1,57 +1,142 @@
+      const rightSideBarList = ontarget.closest('.sidebar__container').lastElementChild.childNodes;
+
+      rightSideBarList.forEach((rightSideBar) => {
+        if(rightSideBar.nodeName != '#text'){
```
여기 조건문을 제거 하기 위해서, forEach 전에 filter함수를 사용해보면 어떨까 싶군요.

crongro commented on index.js:
```diff
@@ -1,57 +1,142 @@
+      dimLayer.style.display = 'none';
+    }
+    if(element === 'account-link') {
+      findSibling(ontarget, 2).style.display = 'none';
+      findSibling(ontarget, 4).style.display = 'none';
```
2,4가 매직넘버 같은 느낌인데.. 없애면 더 좋겠네요.

crongro commented on index.js:
```diff
@@ -1,57 +1,142 @@
+  const findSibling = (node, targetIndex) => {
+    const parent = node.parentNode;
+    const children = parent.childNodes;
+    const index = Array.prototype.indexOf.call(children, node);
```
Array.prototype.indexOf.call 이 코드가 어떻게 동작하는지 설명해보세요! 
그리고 이 방법말고 다른 방법도 있는지 알아보세요.

crongro reviewed (commented):
수고하셨습니다~ 
리뷰 참고하세요!
